lib/cisco_node_utils/syslog_server.rb

```diff
@@ -42,12 +52,20 @@ def initialize(name,
 
     def self.syslogservers
       hash = {}
-
       syslogservers_list = config_get('syslog_server', 'server')
       return hash if syslogservers_list.nil?
 
       syslogservers_list.each do |id|
+        # The YAML regex isn't specific enough for some platforms,
+        # so we have to do further checking.
+        begin
+          IPAddr.new(id)
+        rescue
+          next
+        end
+
         level = config_get('syslog_server', 'level', id)
+        level = LEVEL_TO_NUM[level] if platform == :ios_xr
 
         vrf = config_get('syslog_server', 'vrf', id)
```

Comment:
@glennmatthews Sure, however what would the design be in relation to the pre-existing vrf parameter? Would it be a simple "If vrf is specified, only return the hosts that are part of that vrf"?

So basically, add the following line to line 71:
````
next if !vrf_arg.nil? && vrf_arg != vrf
````

Thanks